src/services/smart-contracts/actions/core.js

```diff
@@ -106,7 +111,134 @@ function getReadyCampaign(campaign, identity, Dai) {
+	const channels = await requester
+		.fetch({
+			route: '/campaigns',
+			method: 'GET',
+			queryParams: {
+				status: STATES.join(','),
+			},
+		})
+		.then(res => res.json())
+
+	return Promise.all(
+		channels
+			.map(channel => {
+				const { lastApprovedBalances } = channel.status
+				if (lastApprovedBalances) {
+					const allLeafs = Object.keys(lastApprovedBalances).map(k =>
+						Channel.getBalanceLeaf(k, lastApprovedBalances[k])
+					)
+					const mTree = new MerkleTree(allLeafs)
+					return { lastApprovedBalances, mTree, channel }
+				} else {
+					return { lastApprovedBalances: null, mTree: null, channel }
+				}
+			})
+			.filter(({ channel, lastApprovedBalances }) => {
+				if (channel.status.name === 'Expired') {
+					return channel.creator === identityAddr
+				}
+				return lastApprovedBalances && !!lastApprovedBalances[identityAddr]
+			})
+			.map(({ channel, lastApprovedBalances }) => ({
+				channel,
+				balance: lastApprovedBalances[identityAddr],
+			}))
+			.sort((c1, c2) => {
+				// Sorting by most balance so we can get top N needed using amountToSweep so we send as few transactions as possible
+				return new BN(c2.status.lastApprovedBalances[identityAddr]).gte(
+					new BN(c1.status.lastApprovedBalances[identityAddr])
+				)
+			})
+	)
+}
+
+async function getChannelsToSweepFrom({ amountToSweep, identityAddr }) {
+	const STATES = [
+		'Active',
+		'Ready',
+		'Exhausted',
+		'Offline',
+		'Unhealthy',
+		'Withdraw',
+		'Expired',
+	]
+
+	const requester = new Requester({ baseUrl: ADEX_MARKET_HOST })
+	const allChannels = await getChannelsWithBalance({
+		identityAddr,
+		requester,
+		STATES,
+	})
+
+	// Could be done with map/reduce but figured this for loop is much simpler in this case
+	const channelsToWithdrawFrom = []
+	const sum = new BN('0')
+	for (let i = 0; i < allChannels.length; i++) {
+		if (sum.gte(new BN(amountToSweep))) {
+			break
+		}
+		const balance = new BN(
+			allChannels[i].status.lastApprovedBalances[identityAddr]
+		)
+		sum.iadd(balance)
+		channelsToWithdrawFrom.push(allChannels[i])
+	}
+
+	return channelsToWithdrawFrom
+}
+
+function getExpiredToWithdraw(channel) {
+	const { lastApprovedBalances } = channel.status
+	const toWithdraw = Object.keys(lastApprovedBalances).reduce((acc, val) => {
+		return acc.sub(new BN(lastApprovedBalances[val]))
+	}, new BN(channel.depositAmount))
+
+	return toWithdraw.toString()
+}
+
+export async function sweepChannels({ feeTokenAddr, account, amountToSweep }) {
+	const { wallet } = account
+	const { provider, Dai, Identity } = await getEthers(wallet.authType)
+	const identityAddr = account.identity.address
+	const channelsToSweep = await getChannelsToSweepFrom({
+		amountToSweep,
+		identityAddr,
+	})
+	const identityContract = new Contract(identityAddr, Identity.abi, provider)
+	const initialNonce = (await identityContract.nonce()).toNumber()
+
+	const txns = channelsToSweep.map((c, i) => {
+		const toWithdraw =
```

Comment:
ok, so your `toWithdraw` is equivalent to the `outstanding` variable in `routineAuthsLoop`
also you need it when sorting but you don't need it here

`channelWithdrawExpired` just takes the entire channel 
`channelWithdraw` takes a bunch of stuff, one of them is the leaf from `lastApprovedBalances`